packages/nextjs/src/utils/config.ts

```diff
@@ -1,94 +1,95 @@
-/* eslint-disable @typescript-eslint/no-non-null-assertion */
-/* eslint-disable @typescript-eslint/no-var-requires */
-/* eslint-disable @typescript-eslint/no-explicit-any */
 import { getSentryRelease } from '@sentry/node';
 import { logger } from '@sentry/utils';
 import defaultWebpackPlugin, { SentryCliPluginOptions } from '@sentry/webpack-plugin';
 import * as SentryWebpackPlugin from '@sentry/webpack-plugin';
-import * as fs from 'fs';
-import * as path from 'path';
-
-/**
- * Starting at `startPath`, move up one directory at a time, searching for `searchFile`.
- *
- * @param startPath The location from which to start the search.
- * @param searchFile The file to search for
- * @returns The absolute path of the file, if it's found, or undefined if it's not
- */
-function findUp(startPath: string, searchFile: string): string | undefined {
-  if (!fs.existsSync(startPath)) {
-    throw new Error(`The given \`startPath\` value (${startPath}) does not exist.`);
-  }
 
-  // if the last segment of `startPath` is a file, trim it off so that we start looking in its parent directory
-  let currentDir = fs.statSync(startPath).isFile() ? path.dirname(startPath) : startPath;
-  // eslint-disable-next-line no-constant-condition
-  while (true) {
-    const possiblePath = path.join(currentDir, searchFile);
-    if (fs.existsSync(possiblePath)) {
-      return possiblePath;
-    }
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type PlainObject<T = any> = { [key: string]: T };
 
-    const parentDir = path.resolve(currentDir, '..');
-    // this means we've gotten to the root
-    if (currentDir === parentDir) {
-      break;
-    }
-    currentDir = parentDir;
-  }
+// Man are these types hard to name well. "Entry" = an item in some collection of items, but in our case, one of the
+// things we're worried about here is property (entry) in an object called... entry. So henceforth, the specific
+// proptery we're modifying is going to be known as an EntryProperty, or EP for short.
 
-  return undefined;
-}
+// The function which is ultimately going to be exported from `next.config.js` under the name `webpack`
+type WebpackExport = (config: WebpackConfig, options: WebpackOptions) => WebpackConfig;
+// type WebpackExport = (config: WebpackConfig, options: WebpackOptions) => Promise<WebpackConfig>;
+
+// The two arguments passed to the exported `webpack` function, as well as the thing it returns
+type WebpackConfig = { devtool: string; plugins: PlainObject[]; entry: EntryProperty };
```

Comment:
Maybe use type `Configuration` which is exported from `webpack`? By the way, it probably should be added to peer dependencies (and also to dev dependencies if the type `Configuration` is used)... It's also good from the perspective of letting users know what version of `webpack` this package expects.

Reply:
Next is in the process of moving from webpack 4 to webpack 5 right now. Worth taking that into account with the dep versioning.